qiskit_dynamics/solvers/fixed_step_solvers.py

```diff
@@ -100,6 +106,55 @@ def take_step(generator, t0, y, h):
         eval_time = t0 + (h / 2)
         return expm(generator(eval_time) * h) @ y
 
+    # ensure the output of rhs_func is a raw array
+    def wrapped_rhs_func(*args):
+        return Array(generator(*args)).data
+
+    return fixed_step_solver_template(
+        take_step, rhs_func=wrapped_rhs_func, t_span=t_span, y0=y0, max_dt=max_dt, t_eval=t_eval
+    )
+
+
+def lanczos_diag_solver(
+    generator: Callable,
+    t_span: Array,
+    y0: Array,
+    max_dt: float,
+    t_eval: Optional[Union[Tuple, List, Array]] = None,
+    k_dim: Optional[int] = None,
+):
+    """Fixed-step size matrix exponential based solver implemented using
+    lanczos algorithm. Solves the specified problem by taking steps of
+    size no larger than ``max_dt``.
+
+    Args:
+        generator: Generator for the LMDE.
+        t_span: Interval to solve over.
+        y0: Initial state.
+        max_dt: Maximum step size.
+        t_eval: Optional list of time points at which to return the solution.
+        k_dim: Integer which specifies the dimension of Krylov subspace used for
+               lanczos iteration. Acts as an accuracy parameter. ``k_dim`` must
+               always be less than dimension of generator.
+
+    Returns:
+        OdeResult: Results object.
+
+    Raises:
+        QiskitError: If ``k_dim`` is greater than dimension of generator
+    """
+
+    dim = generator(0).shape[0]
+    if k_dim is None:
+        k_dim = dim // 4
+    if k_dim > dim:
+        raise QiskitError(f"k_dim value {k_dim} is greater than dimension of generator {dim}")
+
+    def take_step(generator, t0, y, h):
+        eval_time = t0 + (h / 2)
+        # since qiskt-dynamics generator is -1j*H but lanczos only works on hermitian arrays
+        return lanczos_expm(1j * generator(eval_time), y, k_dim, h)
```

Comment:
There is another case for keeping it. If `h` is complex for instance, the method will still work. however, if it is multiplied with the generator, then it is no longer anti-hermitian.

Reply:
As for having a default value, setting a default value at the level of `lanczos_expm` would not have any effect since the high level solver would still need it to be passed. 
On the other hand, setting a default value in `fixed_step_solvers.py`  is probably not a good idea since that would make the signature of  `lanczos_diag`  different from `scipy_expm`

Reply:
yeah makes sense